contracts/external/pCash/AaveV3HoldingsOracle.sol

```diff
@@ -150,8 +157,16 @@ contract AaveV3HoldingsOracle is UnderlyingHoldingsOracle {
         (/* */, uint256 supplyCap) = IPoolDataProvider(POOL_DATA_PROVIDER).getReserveCaps(underlying);
         // Supply caps are returned as whole token values
         supplyCap = supplyCap * UNDERLYING_PRECISION;
-        // If supply cap is zero, that means there is no cap on the pool
-        if (supplyCap == 0) {
+        // This is the returned stored token balance of the aToken
+        oracleData.currentExternalUnderlyingLend = _holdingValuesInUnderlying()[0];
+
+        // Sets a cap on the total deposits
+        if (0 < maxDeposit) {
+            oracleData.maxExternalDeposit =  oracleData.currentExternalUnderlyingLend < maxDeposit ?
+                maxDeposit - oracleData.currentExternalUnderlyingLend :
+                0;
+        } else if (supplyCap == 0) {
```

Comment:
This should not be part of the `if-else` block an should be moved after it(after oracleData.maxExternalDeposit is calculated) in case we decide to have it permanently in production otherwise we could end up in situation when pool in near supply cap but we are not detecting it due to existing `maxDeposit`
```sol
        if (0 < maxDeposit && maxDeposit < oracleData.maxExternalDeposit) {
            oracleData.maxExternalDeposit =  oracleData.currentExternalUnderlyingLend < maxDeposit ?
                maxDeposit - oracleData.currentExternalUnderlyingLend :
                0;
        }
  ```